library/objective-c/EnvoyEngineImpl.h

```diff
@@ -0,0 +1,33 @@
+#import "EnvoyEngine.h"
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Wrapper layer for calling into Envoy's C/++ API.
+@interface EnvoyEngineImpl : NSObject <EnvoyEngine>
+
+/**
+ Run the Envoy engine with the provided config and log level.
+
+ @param config The configuration file with which to start Envoy.
+ @return A status indicating if the action was successful.
+ */
++ (EnvoyStatus)runWithConfig:(NSString *)config;
+
+/**
+ Run the Envoy engine with the provided config and log level.
+
+ @param config The configuration file with which to start Envoy.
+ @param logLevel The log level to use when starting Envoy.
+ @return A status indicating if the action was successful.
+ */
++ (EnvoyStatus)runWithConfig:(NSString *)config logLevel:(NSString *)logLevel;
+
+/// Performs necessary setup after Envoy has initialized and started running.
+/// TODO: create a post-initialization callback from Envoy to handle this automatically.
++ (void)setupEnvoy;
```

Comment:
If this is defined here, should probably also be in the `EnvoyEngine` interface

Reply:
I kept this here because the plan is to get rid of it as soon as we have the post-init callback wired up.

Reply:
Sounds good